src/lti/LTI_Message_Launch.php

```diff
@@ -191,7 +192,11 @@ private function get_public_key() {
         foreach ($public_key_set['keys'] as $key) {
             if ($key['kid'] == $this->jwt['header']['kid']) {
                 try {
-                    return openssl_pkey_get_details(JWK::parseKey($key));
+                    return openssl_pkey_get_details(
+                        JWK::parseKeySet([

```

Comment:
It's a shame that `parseKey` is private in the firebase implementation,

It might be neater here to just parse the keyset then check for the key we are looking for is there, then we can avoid the whole for loop here.
```
try {
    $parsed_key_set = JWK::parseKeySet($_public_key_set);
    if (isset($parsed_key_set[$this->jwt['header']['kid']])) {
        return openssl_pkey_get_details($parsed_key_set[$this->jwt['header']['kid']);
    }
} catch(\Exception $e) {
    throw new LTI_Exception("Unable to parse public key", 1);
}
```

Reply:
It's made public in the latest version thanks to EricTendian :)
https://github.com/firebase/php-jwt/blob/v5.2.0/src/JWK.php#L35